packages/mobile/src/screens/edit-track-screen/EditTrackScreen.tsx

```diff
@@ -235,26 +238,64 @@ export const EditTrackScreen = (props: EditTrackScreenProps) => {
         }
       }
 
+      let streamConditions = metadata.stream_conditions
+      let previewStartSeconds = metadata.preview_start_seconds
+      let isDownloadable = metadata.is_downloadable
+      let isOriginalAvailable = metadata.is_original_available
+
       // If track is usdc gated, then price and preview need to be parsed into numbers before submitting
-      if (isContentUSDCPurchaseGated(metadata.stream_conditions)) {
-        metadata.stream_conditions = {
+      if (isContentUSDCPurchaseGated(streamConditions)) {
+        streamConditions = {
           usdc_purchase: {
-            ...metadata.stream_conditions.usdc_purchase,
+            ...streamConditions.usdc_purchase,
             // Convert dollar price to cents
             // @ts-ignore the price input field stored it as a string that needs to be parsed into a number
             price:
-              Number(metadata.stream_conditions.usdc_purchase.price) *
-              10 ** PRECISION
+              Number(streamConditions.usdc_purchase.price) * 10 ** PRECISION
           }
         }
         // If user did not set usdc gated track preview, default it to 0
-        metadata.preview_start_seconds = Number(
-          metadata.preview_start_seconds ?? 0
+        previewStartSeconds = Number(previewStartSeconds ?? 0)
+
+        // track is downloadable and lossless files are provided by default if track is usdc purchase gated
+        // unless it was already usdc purchase gated
+        const { download_conditions: initialDownloadConditions } =
+          initialValuesProp
+        isDownloadable = !isContentUSDCPurchaseGated(initialDownloadConditions)
```

Comment:
if not purchase gated, we set downloadable to true?

Reply:
yes, remember that we only hit this line if `isContentUSDCPurchaseGated(streamConditions)` is true, meaning it is also download gated.
the comment attempts to explain it, maybe not clear enough?

basically:
- if it was not previously download usdc gated, but now is, then we make sure isDownloadable is true. otherwise, we preserve whatever value the flag previously had.

Reply:
got it. a bit confusing but not blocking at all